test/evm/hardhat/chain-adapters/Arbitrum_Adapter.ts

```diff
@@ -225,4 +252,42 @@ describe("Arbitrum Chain Adapter", function () {
       usdc.address
     );
   });
+  it("Correctly calls the OFT bridge adapter when attempting to bridge USDT", async function () {
+    const internalChainId = arbitrumChainId;
+
+    const { leaves, tree, tokensSendToL2 } = await constructSingleChainTree(usdt.address, 1, internalChainId, 6);
+    await hubPool
+      .connect(dataWorker)
+      .proposeRootBundle([3117], 1, tree.getHexRoot(), consts.mockRelayerRefundRoot, consts.mockSlowRelayRoot);
+    await timer.setCurrentTime(Number(await timer.getCurrentTime()) + consts.refundProposalLiveness + 1);
+
+    // set up correct messenger to be returned on a proper `oftMessengers` call
+    oftAddressBook.oftMessengers.whenCalledWith(usdt.address).returns(oftMessenger.address);
+
+    // set up `quoteSend` return val
+    const msgFeeStruct: MessagingFeeStructOutput = [
+      BigNumber.from(1e9 * 200_000), // nativeFee: 1 GWEI gas price * 200,000 gas cost
+      BigNumber.from(0), // lzTokenFee: 0
+    ] as MessagingFeeStructOutput;
+    oftMessenger.quoteSend.returns(msgFeeStruct);
+
+    // set up `send` return val
+    const msgReceipt: MessagingReceiptStructOutput = [
+      randomBytes32(), // guid
+      BigNumber.from("1"), // nonce
+      msgFeeStruct, // fee
+    ] as MessagingReceiptStructOutput;
+
+    const oftReceipt: OFTReceiptStructOutput = [tokensSendToL2, tokensSendToL2] as OFTReceiptStructOutput;
+
+    oftMessenger.send.returns([msgReceipt, oftReceipt]);
+
+    await hubPool.connect(dataWorker).executeRootBundle(...Object.values(leaves[0]), tree.getHexProof(leaves[0]));
+
+    // Adapter should have approved gateway to spend its ERC20.
+    expect(await usdt.allowance(hubPool.address, oftMessenger.address)).to.equal(tokensSendToL2);
+
+    // We should have called send on the oftMessenger
+    expect(oftMessenger.send).to.have.been.calledOnce;
```

Comment:
I like using the [calledWith](https://smock.readthedocs.io/en/latest/fakes.html#asserting-call-arguments-or-value) syntax to check the arguments passed to `send` are as expected